backend/src/routes/api/nb-events/eventUtils.ts

```diff
@@ -0,0 +1,95 @@
+import { V1EventList } from '@kubernetes/client-node';
+import { KubeFastifyInstance, NotebookStatus } from '../../../types';
+
+export const getNotebookEvents = async (
+  fastify: KubeFastifyInstance,
+  nbName: string,
+): Promise<NotebookStatus> => {
+  const response = await fastify.kube.coreV1Api
+    .listNamespacedEvent(
+      fastify.kube.namespace,
+      undefined,
+      undefined,
+      undefined,
+      `involvedObject.kind=Notebook,involvedObject.name=${nbName}`,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      false,
+    )
+    .then((res) => {
+      return res.body as V1EventList;
+    })
+    .catch((e) => {
+      fastify.log.error(e);
+      return { items: [] } as V1EventList;
+    });
+  let percentile = 0;
+  let status: 'In Progress' | 'Warning' | 'Error' | 'Success' = 'In Progress';
+  const lastItem = response.items[response.items.length - 1];
+  if (lastItem.message.includes('oauth-proxy')) {
```

Comment:
```
  const lastItem = response.items[response.items.length - 1];
  if (lastItem.message.includes('oauth-proxy')) {
```
`lastItem` is undefined if `response.items` has a length of zero.

I think we should look to try and exit with a 404 error instead of optional chaining.

No events should be a NOT FOUND error as it would fail to provide any information of any use.

Reply:
Added a new 404 error in a try catch block encompassing the swtiches and letting the index.ts catch send it. I can make the error more specific to the `lastItem` variable if needed